openedx/metrics-dashboard
# Leaderboard docker-compose devstack [Issue#227]

## Description

Updates https://github.com/openedx/metrics-dashboard/pull/9 to:

* run a working docker container using local configuration
* provide easy-to-use `make` commands
* update documentation


## Supporting information

See:
* https://github.com/edx/open-edx-proposals/issues/226
* https://github.com/edx/open-edx-proposals/issues/227
* [FAL-2172](https://tasks.opencraft.com/browse/FAL-2172) (OpenCraft internal)

## Testing instructions

1. Follow the instructions in the updated `README.md` to [run a local grimoire docker instance](https://github.com/open-craft/metrics-dashboard/blob/jill/e0d/docker-compose/README.md#run-local-instance).
2. Tail `logs/all.log` to watch the progress of ingesting data from the `projects-devstack.json` sources (this will take a little while)
3. Load the Grimoire dashboard, e.g. http://localhost:5601/app/kibana#/dashboard/Overview?_g=() and ensure that you can see the ingested data.
4. Load the Hatstall admin UI, e.g. http://localhost:8000/identities/hatstall/list using the credentials provided in your `.env`.

## Additional information

1. I don't have access to https://github.com/edx/repo-tools-data, which contains [data used to generate the `identities.yaml`](https://github.com/openedx/metrics-dashboard/blob/master/entrypoint.sh#L15) file.
   So the git authors shown in Grimoire are all in the Unknown organization, ungrouped into their respective orgs.
   
   Will resolve this issue in a follow-up task.

To do:
1. fix errors from sirmordred container
2. mark bots.
